webapp/channels/src/components/profile_popover/index.tsx

```diff
@@ -95,40 +56,36 @@ function getDefaultChannelId(state: GlobalState) {
 
 /**
  * The profile popover, or hovercard, that appears with user information when clicking
- * on the username or profile picture of a user.
+ * on the username, profile picture of a user, or others.
+ * However this component should not be used directly, instead use the `ProfilePopoverController`
  */
 const ProfilePopover = ({
-    returnFocus: returnFocusProp,
     userId,
+    userProfileSrc,
     channelId: channelIdProp,
+    hideStatus,
+    fromWebhook,
     hide,
+    returnFocus,
     overwriteIcon,
     overwriteName,
-    src,
-    hideStatus,
-    fromWebhook,
-    ...restProps
-}: ProfilePopoverProps) => {
-    const {formatMessage} = useIntl();
-
+}: Props) => {
     const dispatch = useDispatch();
     const user = useSelector((state: GlobalState) => getUser(state, userId));
     const currentTeamId = useSelector((state: GlobalState) => getCurrentTeamId(state));
     const channelId = useSelector((state: GlobalState) => (channelIdProp || getDefaultChannelId(state)));
-    const isAnyModalOpen = useSelector(getIsAnyModalOpen);
     const isMobileView = useSelector(getIsMobileView);
     const teamUrl = useSelector(getCurrentRelativeTeamUrl);
     const modals = useSelector((state: GlobalState) => state.views.modals);
-    const teammateNameDisplay = useSelector(getTeammateNameDisplaySetting);
     const status = useSelector((state: GlobalState) => getStatusForUserId(state, userId) || UserStatuses.OFFLINE);
     const currentUserTimezone = useSelector(getCurrentTimezone);
     const currentUserId = useSelector(getCurrentUserId);
 
-    const closeButtonRef = useRef<HTMLButtonElement>(null);
     const [loadingDMChannel, setLoadingDMChannel] = useState<string>();
-    const returnFocus = useMemo(() => {
-        if (returnFocusProp) {
-            return returnFocusProp;
+
+    const handleReturnFocus = useMemo(() => {
```

Comment:
Do we actually need this with Floating UI? I thought it handles this itself, and I hope it does since this sort of thing is annoying to have to do ourselves 😛

Reply:
Yes this is not needed but i reason for this i have added in the comment

```
 * Function to call to return focus to the previously focused element when the popover closes.
     * If not provided, the popover will automatically determine the previously focused element
     * and focus that on close. However, if the previously focused element is not correctly detected
     * by the popover, or the previously focused element will disappear after the popover opens,
     * it is necessary to provide this function to focus the correct element.
 returnFocus?: () => void;
 ```